src/infrastructure/database/repositories/history.checks.repository.spec.js

```diff
@@ -0,0 +1,116 @@
+import { SteamApp } from "../../../models/steam.app.js";
+import { initiateInMemoryDatabase } from "../in.memory.database.client.js";
+import { HistoryChecksRepository } from "./history.checks.repository.js";
+
+describe("HistoryChecksRepository", function () {
+  describe(".insertManyHistoryChecks inserts history check documents into the database.", function () {
+    describe("If two documents are inserted,", function () {
+      beforeAll(async function () {
+        this.databaseClient = await initiateInMemoryDatabase(["history_checks"]);
+
+        const historyChecksRepo = new HistoryChecksRepository(this.databaseClient);
+
+        await historyChecksRepo.insertManyHistoryChecks([
+          { id: 1, name: "First Game" },
+          { id: 2, name: "Second Game" },
+        ]);
+
+        this.result = await this.databaseClient.getAll("history_checks");
+      });
+
+      afterAll(function () {
+        this.databaseClient.disconnect();
+      });
+
+      it("the resulting array has a length of 2", function () {
+        expect(this.result.length).toBe(2);
+      });
+
+      it("the first array has the correct values", function () {
+        expect(this.result[0].id).toBe(1);
+        expect(this.result[0].name).toBe("First Game");
+      });
+
+      it("the second array has the correct values", function () {
+        expect(this.result[1].id).toBe(2);
+        expect(this.result[1].name).toBe("Second Game");
+      });
+    });
+  });
+
+  describe(".updateHistoryChecks modifies the provided database documents.", function () {
+    describe("When three history checks are provided", function () {
+      beforeAll(async function () {
+        this.databaseClient = await initiateInMemoryDatabase(["history_checks"]);
+
+        const historyChecksRepo = new HistoryChecksRepository(this.databaseClient);
+
+        await historyChecksRepo.insertManyHistoryChecks([
+          { gameId: 1, checked: false, found: false },
+          { gameId: 2, checked: false, found: false },
+          { gameId: 3, checked: false, found: false },
+        ]);
+
+        const updatedHistoryChecks = [
+          {
+            gameId: 1,
+            checked: true,
+            found: true,
+            source: SteamApp.validDataSources.steamcharts,
+          },
+          {
+            gameId: 2,
+            checked: true,
+            found: false,
+            source: SteamApp.validDataSources.steamcharts,
+          },
+          {
+            gameId: 3,
+            checked: true,
+            found: true,
+            source: SteamApp.validDataSources.steamcharts,
+          },
+        ];
+
+        await historyChecksRepo.updateHistoryChecks(updatedHistoryChecks);
+
+        this.unresolvedResult = this.databaseClient.getAll("history_checks");
```

Comment:
what is this? :)

Reply:
I just wanted to check that what is returned before we await for the result is a promise. I guess it's unnecessary, will remove it.